hudi-common/src/main/java/org/apache/hudi/common/model/HoodieLogFile.java

```diff
@@ -98,7 +111,10 @@ public String getFileExtension() {
   }
 
   public Path getPath() {
-    return new Path(pathStr);
+    if (path == null) {
+      this.path = new Path(pathStr);
```

Comment:
Could you use `CachingPath` instead?

Reply:
@yihua Thanks for your comments and suggestions，I have taken your suggestion and submitted again.  I haven't compared the memory usage before and after the patch in time. But with four new variables we can roughly estimate the  memory usage.

Reply:
This is still a Path and not a CachingPath